code/core/src/core-server/dev-server.ts

```diff
@@ -165,6 +166,34 @@ export async function storybookDevServer(options: Options) {
         return;
       }
     });
+
+    app.get('/manifests/components.html', async (req, res) => {
+      try {
+        const componentManifestGenerator: ComponentManifestGenerator = await options.presets.apply(
+          'experimental_componentManifestGenerator'
+        );
+        const indexGenerator = await initializedStoryIndexGenerator;
+
+        if (!componentManifestGenerator || !indexGenerator) {
+          res.statusCode = 400;
+          res.setHeader('Content-Type', 'text/html; charset=utf-8');
+          res.end(`<pre>No component manifest generator configured.</pre>`);
+          return;
+        }
+
+        const manifest = (await componentManifestGenerator(
+          indexGenerator as unknown as import('storybook/internal/core-server').StoryIndexGenerator
+        )) as ComponentsManifest;
+
+        res.setHeader('Content-Type', 'text/html; charset=utf-8');
+        res.end(renderManifestComponentsPage(manifest));
```

Comment:
instead of generating the manifest here, I would instead have served a static html page that fetched `/manifests/components.json` client side and then built the the DOM from that, client-side. should be fast enough too. the upside of that approach is that you can then make the html manifest available at build time too, without having the manifest MBs twice.

WDYT @shilman ?